sqldelight-gradle-plugin/src/main/kotlin/app/cash/sqldelight/gradle/VerifyMigrationTask.kt

```diff
@@ -51,6 +53,8 @@ abstract class VerifyMigrationTask : SqlDelightWorkerTask() {
 
   @Input var verifyDefinitions: Boolean = true
 
+  @get:Internal abstract var driverInitializer: DriverInitializer
```

Comment:
The build failed: 
This does not work. You either need a default value, like `null`, make it `lateinit` or use a Provider. If you want to use a lambda, you could also use an `Action`, this will work out of the box with Gradle and serializable.

Reply:
lambda would be nice to make the ABI of sqldelight smaller

Reply:
I changed the property to an `Action<Unit>` and left the `VerifyMigrationWorkParameters.driverInitializer` as `Property<Action<Unit>>`

```
  /**
   * Allows consumers to configure and register (with [DriverManager]) their custom drivers prior to
   * running migration verification task.
   */
  @get:Internal abstract var driverInitializer: Action<Unit>
```
```
  interface VerifyMigrationWorkParameters : WorkParameters {
    val workingDirectory: DirectoryProperty
    val projectName: Property<String>
    val properties: Property<SqlDelightDatabaseProperties>
    val compilationUnit: Property<SqlDelightCompilationUnit>
    val verifyMigrations: Property<Boolean>
    val verifyDefinitions: Property<Boolean>
    val driverInitializer: Property<Action<Unit>>
  }
```

but now I'm getting the following Exception:

```
Caused by: org.gradle.internal.snapshot.impl.IsolationException: Could not isolate value app.cash.sqldelight.gradle.VerifyMigrationTask$VerifyMigrationWorkParameters_Decorated@6c2ef097 of type VerifyMigrationTask.VerifyMigrationWorkParameters
        at org.gradle.internal.snapshot.impl.DefaultIsolatableFactory.isolate(DefaultIsolatableFactory.java:51)
        at org.gradle.workers.internal.DefaultActionExecutionSpecFactory.newIsolatedSpec(DefaultActionExecutionSpecFactory.java:48)
        at org.gradle.workers.internal.DefaultWorkerExecutor.submitWork(DefaultWorkerExecutor.java:155)
        ... 125 more
Caused by: org.gradle.internal.snapshot.ValueSnapshottingException: Could not serialize value of type SetupSqldelightKt...
        at org.gradle.internal.snapshot.impl.AbstractValueProcessor.newValueSerializationException(AbstractValueProcessor.java:187)
        at org.gradle.internal.snapshot.impl.AbstractValueProcessor.javaSerialization(AbstractValueProcessor.java:178)
        at org.gradle.internal.snapshot.impl.AbstractValueProcessor.processValue(AbstractValueProcessor.java:157)
        at org.gradle.internal.snapshot.impl.AbstractValueProcessor.processValue(AbstractValueProcessor.java:134)
        at org.gradle.internal.snapshot.impl.AbstractValueProcessor.processValue(AbstractValueProcessor.java:134)
        at org.gradle.internal.snapshot.impl.AbstractValueProcessor.processValue(AbstractValueProcessor.java:121)
        at org.gradle.internal.snapshot.impl.AbstractValueProcessor.processValue(AbstractValueProcessor.java:134)
        at org.gradle.internal.snapshot.impl.DefaultIsolatableFactory.isolate(DefaultIsolatableFactory.java:49)
        ... 127 more
Caused by: java.io.NotSerializableException: SetupSqldelightKt$setupSqldelight$1$2$1
        at org.gradle.internal.snapshot.impl.AbstractValueProcessor.javaSerialization(AbstractValueProcessor.java:175)
        ... 133 more

```

perhaps I'm not using the `Action` appropriately?

Reply:
maybe just do the default null thing then

Reply:
The error comes from the action. Wanna try it again without `Property<Action<Unit>>` but just `Action<Unit>?

Reply:
This would require making one of the `WorkParameters` a var, wasn't sure if that was OK. Let me try

Reply:
@hfhbd still getting a similar error:

```
Caused by: org.gradle.internal.snapshot.impl.IsolationException: Could not isolate value app.cash.sqldelight.gradle.VerifyMigrationTask$VerifyMigrationWorkParameters_Decorated@5a379943 of type VerifyMigrationTask.VerifyMigrationWorkParameters
        at org.gradle.internal.snapshot.impl.DefaultIsolatableFactory.isolate(DefaultIsolatableFactory.java:51)
        at org.gradle.workers.internal.DefaultActionExecutionSpecFactory.newIsolatedSpec(DefaultActionExecutionSpecFactory.java:48)
        at org.gradle.workers.internal.DefaultWorkerExecutor.submitWork(DefaultWorkerExecutor.java:155)
        ... 125 more
Caused by: org.gradle.internal.snapshot.ValueSnapshottingException: Could not serialize value of type SetupSqldelightKt...
        at org.gradle.internal.snapshot.impl.AbstractValueProcessor.newValueSerializationException(AbstractValueProcessor.java:187)
        at org.gradle.internal.snapshot.impl.AbstractValueProcessor.javaSerialization(AbstractValueProcessor.java:178)
        at org.gradle.internal.snapshot.impl.AbstractValueProcessor.processValue(AbstractValueProcessor.java:157)
        at org.gradle.internal.snapshot.impl.AbstractValueProcessor.processValue(AbstractValueProcessor.java:121)
        at org.gradle.internal.snapshot.impl.AbstractValueProcessor.processValue(AbstractValueProcessor.java:134)
        at org.gradle.internal.snapshot.impl.DefaultIsolatableFactory.isolate(DefaultIsolatableFactory.java:49)
        ... 127 more
Caused by: java.io.NotSerializableException: SetupSqldelightKt$setupSqldelight$1$2$1
        at org.gradle.internal.snapshot.impl.AbstractValueProcessor.javaSerialization(AbstractValueProcessor.java:175)
        ... 131 more
```

```
@get:Internal abstract var driverInitializer: Action<Unit>

  interface VerifyMigrationWorkParameters : WorkParameters {
    val workingDirectory: DirectoryProperty
    val projectName: Property<String>
    val properties: Property<SqlDelightDatabaseProperties>
    val compilationUnit: Property<SqlDelightCompilationUnit>
    val verifyMigrations: Property<Boolean>
    val verifyDefinitions: Property<Boolean>
    var driverInitializer: Action<Unit>
  }
```